src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/EventPipeSession.cs

```diff
@@ -30,16 +29,84 @@ private EventPipeSession(IpcEndpoint endpoint, IpcResponse response, ulong sessi
 
         internal static EventPipeSession Start(IpcEndpoint endpoint, EventPipeSessionConfiguration config)
         {
-            IpcMessage requestMessage = CreateStartMessage(config);
+            IpcMessage requestMessage = CreateStartMessage(config, EventPipeCommandId.CollectTracing4);
```

Comment:
As an optimization, you don't have to always start with `CollectTracing4`. 
`CollectTracing4` is currently only necessary for gc-collect trace and nothing else. This will lead to an unnecessary retry for all old runtimes for all (but gc-collect) usages.